EFCache/CachingCommand.cs

```diff
@@ -173,24 +173,32 @@ public override bool DesignTimeVisible
 
         protected override DbDataReader ExecuteDbDataReader(CommandBehavior behavior)
         {
+            object cacheLock = null;
+			var affectedEntitySets = _commandTreeFacts.AffectedEntitySets.Select(s => s.Name).ToList();
             if (!IsCacheable)
             {
-                var result = _command.ExecuteReader(behavior);
-
                 if (!_commandTreeFacts.IsQuery)
                 {
+                    cacheLock = _cacheTransactionHandler.Lock(affectedEntitySets);

```

Comment:
@jkdufair so we do not lock the cache on reads right now? doesn't the "getItem" method still modify the cache with expiry updates? Or is that a TODO since we aren't using any expiring keys yet?

Reply:
It's a TODO. See line 256 😄

Reply:
@jkdufair I did see the TODO, but it is in the other project ;)

Reply:
@builtbywill Oh lol of course. Might not even be worth it to lock on expiry actually.